### PR TITLE
Added missing drivers for piHat

### LINKED ISSUE #260: Main script doesn't run without a pi hat
**Describe the bug**
The main script works only with pi hat , but throws as error without it and has some errors with snowboy refactor

**Expected behavior**
To make it work without it again

**Would you like to work on it?**
Yes

EDIT
Error Log
screenshot 2018-07-15 at 11 46 58 pm

### PR DESCRIPTION
**Pull request by stealthanthrax**
Fixes #260 

Added drivers with piHat

### CODE REVIEW

**Comment by stealthanthrax** [comment]
@woshikie , it was discussed in an earlier meeting that we should go with this approach , also these are just drivers for the respeaker and without it the code will crash out

**Comment by woshikie** [comment]
@stealthanthrax 
Why should a user be installing seeed's respeaker's drivers if the user is not using it?

**Comment by woshikie** [comment]
why not change the code to not crash when ran without the driver?

**Comment by stealthanthrax** [comment]
@woshikie , we'll have to add try and except statements everywhere

**Comment by woshikie** [comment]
you have a class wrapper for the leds in `/main/states/led.py`
you can modify it such that if you detect that respeaker is not installed, the initialization function is stopped and every other function will also not work if initialization is failed

**Comment by Orbiter** [comment]
Well I understand that with this change susi_linux will install piHat and will work even if no respeaker is attached and/or used. Is that correct? If yes, my approval is valid. If no I would think about it again ;)

**Comment by stealthanthrax** [comment]
Yes @Orbiter , it will work without pi hat attached

**Comment by stealthanthrax** [comment]
Updated @hongquan

File: install.sh [hunk 1]
```diff
@@ -59,6 +59,16 @@ function install_swig_from_sources()
     rm -rf swig-3.0.12*
 }
 
+function install_seed_voicecard_driver()
+{
+    echo "installing Pi hat drivers from source"
```
Comment (hongquan) [inline install.sh]: Fix this line, too.

**Comment by stealthanthrax** [comment]
@hongquan , updated

File: install.sh [hunk 2]
```diff
@@ -78,6 +88,7 @@ function install_dependencies()
         if ! [ -x "$(command -v swig)" ]
         then
             install_swig_from_sources
+            install_seed_voicecard_driver
```
Comment (woshikie) [inline install.sh]: Why is `install_seed_voicecard_driver` called from within `install_flite_from_source`? 
It should be called from the main thread

Comment (hongquan) [reply install.sh]: @woshikie It is not called from `install_flite_from_source`. You are cheated by the "diff" view.

**Pull request merged**